iroh/src/magicsock/relay_actor.rs

```diff
@@ -274,9 +273,24 @@ impl ActiveRelayActor {
     async fn run(mut self) -> anyhow::Result<()> {
         inc!(MagicsockMetrics, num_relay_conns_added);
 
-        loop {
-            let Some(client) = self.run_dialing().instrument(info_span!("dialing")).await else {
-                break;
+        let mut backoff = ExponentialBuilder::new()
+            .with_min_delay(Duration::from_millis(10))
+            .with_max_delay(Duration::from_secs(5))
+            .without_max_times()
+            .build();
+
+        'run: loop {
+            let client = 'dial: loop {
+                match self.run_dialing().instrument(info_span!("dialing")).await {
+                    None => break 'run,
+                    Some(Ok(client)) => break 'dial client,
+                    Some(Err(err)) => {
+                        let sleep = backoff.next().expect("backoff is infinite");
```

Comment:
I know that's true, but I think our style in iroh has been to try and avoid this and instead let such mistakes be handled by errors.  This `.run()` method already returns an error so might as well return this as an error?  I think in general our panic propagation is not good enough that I'd be comfortable with this approach.  Currently they often get lost on a task boundary.  Plus, as a library in possibly foreign code it's not very nice to panic on folks.

Reply:
:+1: Changed to return an error instead.